src/features/documentSymbolProvider.ts

```diff
@@ -46,11 +46,49 @@ function createSymbols(elements: Structure.CodeElement[]): vscode.DocumentSymbol
     return results;
 }
 
-function createSymbolForElement(element: Structure.CodeElement): vscode.DocumentSymbol {
+function getNameAndDetailsForElement(element: Structure.CodeElement, parentElement?: Structure.CodeElement): { name: string, details: string } {
+    switch (element.Kind) {
+        case SymbolKinds.Class:
+        case SymbolKinds.Delegate:
+        case SymbolKinds.Enum:
+        case SymbolKinds.Interface:
+        case SymbolKinds.Struct:
+        case SymbolKinds.Namespace:
+            if (typeof parentElement === 'undefined') {
+                return { name: element.DisplayName, details: '' };
+            }
+            const prefix = `${parentElement.DisplayName}.`;
+            if (!element.DisplayName.startsWith(prefix)) {
+                return { name: element.DisplayName, details: '' };
+            }
+            const name = element.DisplayName.slice(prefix.length);
```

Comment:
I think that in most cases the `element.Name` property should be the non-fully qualified name of the class/type/interace/etc.

e.g.
<img width="492" alt="image" src="https://user-images.githubusercontent.com/5749229/216137596-b32502b6-a027-4ac9-a147-e545a4f0a7b4.png">


So I think we might not need to do any slicing here, instead just pass in the 'name' as the as the name and the displayName as the details

Reply:
Thanks for the review, nw about the delay :)

The problem is with namespace elements - without the slice the namespace `Foo.Bar.Baz` (without any parents) will resolve as `Baz`, which causes some confusion with the breadcrumbs, making it seem like the namespace is just Baz.

Reply:
I see what you mean, looks like the server just populates the name with the last identifier.  

Can we do this:
1.  Limit the slicing to only happen for namespace case
2.  For the rest, use the name and display name as-is.

We could change the server side to fix this, but seems like overkill for such a small change, so I'd rather not.

Reply:
Hey - I've done that, with a slight tweak - I've used `DisplayName` for the name for some types, and `Name` for others, so that more information is given - for example, below each function's `Name` is just `OverloadMethod`, which is a bit worse IMO in the outliner on small sizes (and gives less information in the breadcrumb). Having it be the `DisplayName` is how it is currently. It's up to you though, if you think the `Name` version is better I can change it to that.

### `DisplayName`
![image](https://user-images.githubusercontent.com/4435960/216698789-d8fc28e4-e6ed-4e53-93b2-55b54d4631f7.png)

### `Name`
![image](https://user-images.githubusercontent.com/4435960/216699110-8a4a0ea5-aef5-46cb-89a7-02cf7eabf1f3.png)

Reply:
Ah absolutely, using DisplayName there looks better - good call.